src/bin/cargo-msrv.rs

```diff
@@ -255,21 +266,16 @@ enum TracingGuard {
     None,
 }
 
-fn log_folder() -> Result<PathBuf, InstanceError> {
+fn log_folder() -> Result<PathBuf, SetupError> {
     dirs::data_local_dir()
         .map(|path| path.join("cargo-msrv"))
-        .ok_or(InstanceError::UnableToAccessLogFolder)
+        .ok_or(SetupError::UnableToAccessLogFolder)
 }
 
+/// Error which occurred during setting up the program or shutting it down.
+/// Does not cover errors which occur during execution.
 #[derive(Debug, thiserror::Error)]
-enum InstanceError {
-    // Only for compat. with `Context::try_from`, which is not as easily converted to this Error type
-    // of the bin crate.
-    // For that reason, we do not derive `#[from] CargoMSRVError`, so we don't silently miss calls
-    // which may produce an `Err(CargoMSRVError)` where we do not want it.
-    #[error("{0}")]
-    CargoMsrv(CargoMSRVError),
-
+enum SetupError {
```

Comment:
If you want I can revert this again to `InstanceError` or any other name you think fits better.